staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -325,65 +333,67 @@ func (o *DebugOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
 	}
 	o.podClient = clientset.CoreV1()
 
-	r := f.NewBuilder().
+	builder := f.NewBuilder().
 		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
 		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.TargetNames...).
-		Do()
-	if err := r.Err(); err != nil {
-		return err
+		SingleResourceType().
+		Flatten()
+	if len(o.FilenameOptions.Filenames) > 0 {
+		builder.FilenameParam(o.explicitNamespace, &o.FilenameOptions)
 	}
 
-	err = r.Visit(func(info *resource.Info, err error) error {
-		if err != nil {
-			// TODO(verb): configurable early return
-			return err
-		}
+	infos, err := builder.Do().Infos()
+	if err != nil {
+		return err
+	}
 
-		var (
-			debugPod      *corev1.Pod
-			containerName string
-			visitErr      error
-		)
-		switch obj := info.Object.(type) {
-		case *corev1.Node:
-			debugPod, containerName, visitErr = o.visitNode(ctx, obj)
-		case *corev1.Pod:
-			debugPod, containerName, visitErr = o.visitPod(ctx, obj)
-		default:
-			visitErr = fmt.Errorf("%q not supported by debug", info.Mapping.GroupVersionKind)
-		}
-		if visitErr != nil {
-			return visitErr
-		}
+	if len(infos) != 1 {
+		klog.V(4).Infof("Objects: %#v", infos)
+		return fmt.Errorf("you can debug only one resource at a time")
```

Comment:
why disallow debugging multiple objects?

Reply:
Probably I made a mistake by assuming that `kubectl debug` supports only one resource at a time. All other changes I made is to guarantee that. Does that mean `kubectl debug` support multiple debugging?. What happens if user pass `-it` flag in multiple resources?

Reply:
After thinking a little bit; there is no way passing multiple resources in `kubectl debug` by name. Up to now, we always support one resource already. But with `-f` flag, users can pass multiple resources and I did these changes to prevent that.

Reply:
iirc, we always supported multiple targets in `kubectl debug`. In the case of `-it` kubectl would proceed in order, connecting to the terminal and proceeding to the next target when the connection returned.

Reply:
Sounds good, thanks for clarification :)